### PR TITLE
modify: Change history_utils to handle /api/history 2.0

### LINKED ISSUE #106: feat: use new history files for 14.0+ history
History will be now stored in https://downloads.keyman.com/history/HISTORY.md. Just the one file. The format of the file, is approximately:

```
# Keyman Version History

## <version> <tier> <yyyy-mm-dd>
<type>[(<scope>[/<sub-scope>)]: title [(#<prnum>)]
```

For example:

```
# Keyman Version History

## 14.0.4 alpha 2020-02-03

chore(ci): trigger builds after version increment (#2572)
```

We need to parse this for presenting history to users for version 14.0 and later. A single history view covering all platforms is appropriate -- sort the entries by scope to make it easier to read.

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #106 by making use of the new downloads.keyman.com/history API 2.0 from keymanapp/downloads.keyman.com#30

When API 2.0 (displays Keyman 14.0+ history) is used, history_utils also calls API 1.0 to include the older history.

### CODE REVIEW

File: _includes/includes/history_utils.php [hunk 1]
```diff
@@ -9,34 +9,64 @@
...
 
-  // Definition of a few adjustments to our history.md spec for nicer website display
+    // Adjust the publish dates of API 2.0 for displaying on website
+    $regex_dated_version_src = "/(\d+.\d+(.\d+)? (?:stable|beta|alpha)) (\d{4}-\d{2}-\d{2})/"; // (\d+.\d+.\d+) [stable|beta|alpha]
+    $regex_dated_version_dst = "\${1}\n Published \${3}.";
+
+    // The actual replacements.
+    //$contents = preg_replace($regex_header_line, "", $contents);
+    $contents = preg_replace($regex_dated_version_src, $regex_dated_version_dst, $contents);
+
```
Comment (mcdurdin) [inline _includes/includes/history_utils.php:51]:
```suggestion

    // Tighten formatting -- h3, remove redundant paragraphs
    $contents = preg_replace('/^## /m', '### ', $contents);
    $contents = preg_replace("/\n\n\\* /", "\n* ", $contents);
```

Just a suggestion 😉

File: developer/engine/web/14.0/history.php [hunk 2]
```diff
@@ -17,7 +17,7 @@
```
Comment (mcdurdin) [inline developer/engine/web/14.0/history.php:17]: Should this just be a redirect to web/history?

File: developer/14.0/main/history.php [hunk 3]
```diff
@@ -17,7 +17,7 @@
```
Comment (mcdurdin) [inline developer/14.0/main/history.php:17]: Should this be a redirect to developer/version-history?

**Review by mcdurdin — APPROVED** [review]

File: developer/14.0/main/history.php (hunk 3 above)
Comment (darcywong00) [reply developer/14.0/main/history.php]: added to a new web.config. Please re-squiz

Comment (darcywong00) [reply developer/14.0/main/history.php]: well, that went horribly awry (broken links). I just changed the history links instead

**Review by mcdurdin — APPROVED** [review]
LGTM

**Pull request merged**